nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -1998,7 +1998,16 @@ def on_load_checkpoint(self, checkpoint) -> None:
                         key.replace('model.', ''): checkpoint_state_dict.pop(key)
                         for key in list(checkpoint_state_dict.keys())
                     }
-                    module.load_state_dict(checkpoint_state_dict, strict=True)
+                    try:
+                        module.load_state_dict(checkpoint_state_dict, strict=True)
+                    except Exception as e:
```

Comment:
We could consider explicitly catching just `RuntimeError` that is actually thrown here for missing key(s).

Currently we'll retry on any problem while we just expect to have a workaround for that keys error

Reply:
@janekl, you're right, thanks. I've just forgot to change it before pushing.